src/modules/page/admin/main.php

```diff
@@ -24,69 +23,59 @@
+}
 
-        $xtpl->parse('main.row.status');
-    }
+$tpl = new \NukeViet\Template\NVSmarty();
+$tpl->registerPlugin('modifier', 'nformat', 'nv_number_format');
+$tpl->registerPlugin('modifier', 'dformat', 'nv_datetime_format');
+$tpl->setTemplateDir(get_module_tpl_dir('main.tpl'));
+$tpl->assign('LANG', $nv_Lang);
+$tpl->assign('MODULE_NAME', $module_name);
+$tpl->assign('PAGE_CONFIG', $page_config);
 
-    // Chức năng copy
-    if ($page_config['copy_page'] == 1) {
-        $url_copy = NV_BASE_ADMINURL . 'index.php?' . NV_LANG_VARIABLE . '=' . NV_LANG_DATA . '&amp;' . NV_NAME_VARIABLE . '=' . $module_name . '&amp;' . NV_OP_VARIABLE . '=content&amp;copy=1&amp;id=' . $row['id'];
-        $xtpl->assign('URL_COPY', $url_copy);
-        $xtpl->parse('main.row.copy_page');
-    }
+$array_row = [];
+$iw = 0;
+$is_delCache = false;
 
-    $is_excdata = 0;
-    if ($global_config['idsite'] > 0 and isset($site_mods['excdata']) and isset($push_content['module'][$module_name]) and $row['status'] == 1) {
-        $count = $db->query('SELECT COUNT(*) FROM ' . NV_PREFIXLANG . '_' . $site_mods['excdata']['module_data'] . '_sended WHERE id_content=' . $row['id'] . ' AND module=' . $db->quote($module_name))
-            ->fetchColumn();
-        $is_excdata = 1;
-        $row['url_send'] = NV_BASE_ADMINURL . 'index.php?' . NV_LANG_VARIABLE . '=' . NV_LANG_DATA . '&amp;' . NV_NAME_VARIABLE . '=excdata&amp;' . NV_OP_VARIABLE . '=send&amp;module=' . $module_name . '&amp;id=' . $row['id'];
+foreach ($_rows as $row) {
+    ++$iw;
+
+    if ($iw != $row['weight']) {
+        $row['weight'] = $iw;
+        $sth = $db->prepare('UPDATE ' . NV_PREFIXLANG . '_' . $module_data . ' SET weight=' . $row['weight'] . ' WHERE id= :id');
+        $sth->bindParam(':id', $row['id'], PDO::PARAM_STR);
+        $sth->execute();
+        $is_delCache = true;
     }
 
-    $row['edit_time'] = nv_datetime_format($row['edit_time'], 1);
-    $row['add_time'] = nv_datetime_format($row['add_time'], 1);
-
-    $xtpl->assign('ROW', $row);
-
-    if ($is_excdata) {
-        $xtpl->parse('main.row.excdata');
-    }
+    $array_row[] = [
+        'id' => $row['id'],
+        'title' => $row['title'],
+        'weight' => $row['weight'],
+        'hitstotal' => nv_number_format($row['hitstotal']),
+        'checkss' => md5($row['id'] . NV_CHECK_SESSION),
+        'url_view' => NV_BASE_SITEURL . 'index.php?' . NV_LANG_VARIABLE . '=' . NV_LANG_DATA . '&amp;' . NV_NAME_VARIABLE . '=' . $module_name . '&amp;' . NV_OP_VARIABLE . '=' . $row['alias'] . $global_config['rewrite_exturl'],
+        'url_edit' => NV_BASE_ADMINURL . 'index.php?' . NV_LANG_VARIABLE . '=' . NV_LANG_DATA . '&amp;' . NV_NAME_VARIABLE . '=' . $module_name . '&amp;' . NV_OP_VARIABLE . '=content&amp;id=' . $row['id'],
+        'status' => $row['status'],
+        'edit_time' => nv_datetime_format($row['edit_time'], 1),
+        'add_time' => nv_datetime_format($row['add_time'], 1)
+    ];
+
+    $weight_list[] = $row['weight'];
```

Comment:
@nambmtk1993 không cần thiết sinh ra biến `$weight_list` này. Trong tpl em chỉ cần dùng vòng lặp for chạy từ 1 đến `count($DATA)` là xong